File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int longest = 0;
        unordered_set<int> nums_set(nums.begin(), nums.end());
        unordered_set<int> seen;

        for(int num: nums){
            if (seen.find(num) != seen.end()){
                continue;
            }

            int current_longest = 1;
            int current_num = num;
            bool goLeft = true;
            bool goRight = true;
            seen.insert(current_num);

            while (nums_set.find(current_num - 1) != nums_set.end()) {
                current_longest++;
                current_num = current_num - 1;
                seen.insert(current_num);
            }
            current_num = num;
            while (nums_set.find(current_num + 1) != nums_set.end()) {
                current_longest++;
                current_num = current_num + 1;
                seen.insert(current_num);
            }

            if (current_longest > longest) {
                longest = current_longest;
            }
        }
        return longest;
    }
};

File: Data Structures & Algorithms/remove-element/submission-0.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        k = 0
        for i, num in enumerate(nums):
            if num == val:
                nums[i] = None
            else:
                k += 1
        n = len(nums)
        l, r = 0, 1
        while r < n:
            if nums[l] == None:
                while r < n and nums[r] == None:
                    r += 1
                if r < n:
                    tmp = nums[l]
                    nums[l] = nums[r]
                    nums[r] = tmp
            l += 1
            r += 1
        return k
